test/eventsource/eventsource-connect.js

```diff
@@ -199,4 +199,25 @@ describe('EventSource - received response must have content-type to be text/even
 
     assert.strictEqual(onerrorCalls.length, 3)
   })
+
+  test('should try to connect again if server is unreachable', async () => {
+    const domain = 'bad.n' + randomInt(1e10).toString(36) + '.proxy'
+
+    const eventSourceInstance = new EventSource(`http://${domain}`, {
+      node: {
+        reconnectionTime: 1000
+      }
+    })
+
+    const onerrorCalls = []
+    eventSourceInstance.onerror = (error) => {
+      onerrorCalls.push(error)
+    }
+
+    await new Promise(resolve => setTimeout(resolve, 5000))
```

Comment:
This test looks like it will be very flaky

Reply:
See the test above

Reply:
I don't understand that test - it gives 8 seconds for 3 connection errors despite the default reconnection time being  3 seconds? I would expect anything that relies on timing to be flaky, especially when the timing is strict, such as with this test. You give exactly 5 seconds for 5 reconnection errors with a 1 second reconnection time.